eos/eos.hpp

```diff
@@ -448,6 +448,137 @@ template <typename T> class RelativisticEOS {
   Real cl_, cl2_;
 };
 
+template <typename T> class EAPRampEOS {
+  
+  // calculate alpha given a particular pressure
+  PORTABLE_FUNCTION
+  Real alpha_p(const Real P) const {
+    if (P < Pe_) {
+      return alpha0_;
+    } else if (P > Pc_) {
+      return 1.0;
+    } else {
+      return alpha0_ + (P - Pe_) / (Pc_ - Pe_) * (1.0 - alpha0_);
+    }
+  }
+  
+  template<typename F, typename ... ARGS>
+  PORTABLE_FUNCTION
+  auto alpha_impl(F func, const Real rho, ARGS... args) const {
+    constexpr const Real a_eps {1.e-12};
+    Real a {0.0};
+    Real a_old {alpha0_};
+    Real diff {0.0};
+    Real P {0.0};
+    for (int i {0}; i < 10; ++i) {
+      P = *func(a_old*rho, args...) / a_old;
```

Comment:
I think you dropped this `std::forward<Args>(args)...`

Reply:
No more variadic template. Code is now less template-y and more understandable.